tests/tools/utils.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+import json
+import os
+import re
+import sys
+
+import boto3
+import botocore.client
+
+
+def resource_has_tags(resource, tags={}):
+    """Checks if a resource defines all `tags`."""
+    return all(
+        tag_key in resource.keys() and resource[tag_key] == tag_value
+        for (tag_key, tag_value) in tags.items()
+    )
+
+
+class S3ResourceFetcher:
+    """A class for fetching vmm-reference test resources from S3."""
+
+    def __init__(
+            self,
+            resource_bucket,
+            resource_manifest_path,
+            download_location=None,
+    ):
+        """Initialize S3 client, manifest of test resources and S3 bucket name."""
+        with open(resource_manifest_path) as json_file:
+            self._resource_manifest = json.load(json_file)
+        self._latest_version = self.get_latest_version()
+        self._resource_bucket = resource_bucket
+        self._s3 = boto3.client(
+            's3',
+            config=botocore.client.Config(signature_version=botocore.UNSIGNED)
+        )
+        if download_location is None:
+            path = os.path.join(
+                os.path.dirname(__file__),
+                "../../resources")
+            self._default_download_location = os.path.abspath(path)
+        else:
+            self._default_download_location = download_location
+
+        print(
+            "Setting default download location: ", self._default_download_location,
+            file=sys.stderr)
```

Comment:
Instead of `print`ing to `stderr`, you can use the `logging` module which sends `print`s to `stdout` but also has builtin functions that emit logs to `stderr` by default.

```python
log = logging.getLogger(__name__)
log.warning('this goes to stderr')
print('this goes to stdout')
```